packages/admin/cms-admin/src/generator/generateGrid.ts

```diff
@@ -0,0 +1,438 @@
+                    })
+                    .join(" ")}]`;
+
+                return {
+                    name: field.name,
+                    description: field.description,
+                    type,
+                    gridType: "singleSelect" as const,
+                    valueOptions,
+                };
+            }
+            //TODO suppoort n:1 relation with singleSelect
+
+            return {
+                name: field.name,
+                description: field.description,
+                type,
+                gridType,
+                renderCell,
+                valueGetter,
+            };
+        });
+
+    const out = `import { useApolloClient, useQuery } from "@apollo/client";
+    import {
+        CrudContextMenu,
+        GridFilterButton,
+        muiGridFilterToGql,
+        muiGridSortToGql,
+        StackLink,
+        Toolbar,
+        ToolbarAutomaticTitleItem,
+        ToolbarFillSpace,
+        ToolbarItem,
+        useBufferedRowCount,
+        useDataGridRemote,
+        usePersistentColumnState,
+    } from "@comet/admin";
+    import { Add as AddIcon, Edit } from "@comet/admin-icons";
+    import { BlockPreviewContent } from "@comet/blocks-admin";
+    import { Alert, Button, Box, IconButton } from "@mui/material";
+    import { DataGridPro, GridColDef, GridToolbarQuickFilter } from "@mui/x-data-grid-pro";
+    import { useContentScope } from "@src/common/ContentScopeProvider";
+    import {
+        GQL${classNamePlural}GridQuery,
+        GQL${classNamePlural}GridQueryVariables,
+        GQL${classNamePlural}ListFragment,
+        GQLCreate${entityName}Mutation,
+        GQLCreate${entityName}MutationVariables,
+        GQLDelete${entityName}Mutation,
+        GQLDelete${entityName}MutationVariables 
+    } from "./${classNamePlural}Grid.generated";
+    import gql from "graphql-tag";
+    import * as React from "react";
+    import { FormattedMessage, useIntl } from "react-intl";
+    ${Object.entries(rootBlocks)
+        .map(([rootBlockKey, rootBlock]) => `import { ${rootBlock.name} } from "${rootBlock.import}";`)
+        .join("\n")}
+
+    const ${instanceNamePlural}Fragment = gql\`
+        fragment ${classNamePlural}List on ${entityName} {
+            id
+            ${gridColumnFields.map((field) => field.name).join("\n")}
+        }
+    \`;
+    
+    const ${instanceNamePlural}Query = gql\`
+        query ${classNamePlural}Grid($offset: Int, $limit: Int${hasSort ? `, $sort: [${entityName}Sort!]` : ""}${
+        hasSearch ? `, $search: String` : ""
+    }${hasFilter ? `, $filter: ${entityName}Filter` : ""}${hasScope ? `, $scope: ${entityName}ContentScopeInput!` : ""}) {
+            ${gridQuery}(offset: $offset, limit: $limit${hasSort ? `, sort: $sort` : ""}${hasSearch ? `, search: $search` : ""}${
+        hasFilter ? `, filter: $filter` : ""
+    }${hasScope ? `, scope: $scope` : ""}) {
+                nodes {
+                    ...${entityName}sList
```

Comment:
```suggestion
                    ...${entityName}List
```

Reply:
not that easy, it should be:
- ProductsList
- NewsList